src/app/src/components/FacilityListItemsTable.jsx

```diff
@@ -73,251 +73,285 @@ const createSelectedStatusChoicesFromParams = (params) => {
+                    listID,
+                },
+            },
+            history: {
+                push,
+                location: {
+                    search,
+                },
+            },
+        } = this.props;
+        const {
             page,
-            getValueFromEvent(e),
-            params,
-        ));
-        fetchListItems(listID, page, getValueFromEvent(e), params);
-    };
-
-    const handleChangeStatusFilter = (selected) => {
-        const newParams = update(params, {
-            status: { $set: selected ? selected.map(x => x.value) : null },
-        });
-        push(makePaginatedFacilityListItemsDetailLinkWithRowCount(
-            listID,
-            1,
             rowsPerPage,
-            newParams,
-        ));
-        fetchListItems(listID, 1, rowsPerPage, newParams);
-    };
+        } = createPaginationOptionsFromQueryString(search);
 
-    const handleShowAllClicked = () => {
-        const newParams = update(params, {
-            $unset: ['status'],
-        });
-        push(makePaginatedFacilityListItemsDetailLinkWithRowCount(
-            listID,
-            1,
-            rowsPerPage,
-            newParams,
-        ));
-        fetchListItems(listID, 1, rowsPerPage, newParams);
-    };
+        const params = createParamsFromQueryString(search);
 
+        const handleChangePage = (_, newPage) => {
+            push(makePaginatedFacilityListItemsDetailLinkWithRowCount(
+                listID,
+                (newPage + 1),
+                rowsPerPage,
+                params,
+            ));
+            fetchListItems(listID, newPage + 1, rowsPerPage, params);
+        };
 
-    const paginationControlsRow = (
-        <Grid
-            container
-            direction="row"
-            justify="space-between"
-            alignItems="center"
-        >
-            <Grid
-                item
-                sm={12}
-                md={6}
-                style={facilityListItemsTableStyles.summaryStatusStyles}
-            >
-                {makeFacilityListSummaryStatus(list.statuses)}
-            </Grid>
+        const handleChangeRowsPerPage = (e) => {
+            push(makePaginatedFacilityListItemsDetailLinkWithRowCount(
+                listID,
+                page,
+                getValueFromEvent(e),
+                params,
+            ));
+            fetchListItems(listID, page, getValueFromEvent(e), params);
+        };
+
+        const handleChangeStatusFilter = (selected) => {
+            const newParams = update(params, {
+                status: { $set: selected ? selected.map(x => x.value) : null },
+            });
+            push(makePaginatedFacilityListItemsDetailLinkWithRowCount(
+                listID,
+                1,
+                rowsPerPage,
+                newParams,
+            ));
+            fetchListItems(listID, 1, rowsPerPage, newParams);
+        };
+
+        const handleShowAllClicked = () => {
```

Comment:
Now that this is a class component I think it'd make sense to move these functions out of `render` and make them class methods. I think they could just be cut and paste, and then callers updated to call them with `this.whatever`.

Reply:
I did some reading and discovered that defining closures inside `render` like this is a performance drag because since redefining these functions every time  triggers unnessesary rendering. Thanks for suggestion.

Reply:
Addressed in fixup 02efe99